content/exchange/artifacts/Linux.Forensics.EnvironmentVariables.yaml

```diff
@@ -0,0 +1,76 @@
+  This artifact focuses on identifying such modifications by scanning for environment variable alterations, network utilities, and scripting languages commonly used in attacks.  
+  The globs for file paths can be added using a comma-separated list, and the filters are regex patterns that can be modified as needed to tailor the artifact's detection capabilities.
+  
+  **MITRE ATT&CK**: [T1546.004](https://attack.mitre.org/techniques/T1546/004/)
+
+reference:
+  - https://unit42.paloaltonetworks.com/unit42-new-iotlinux-malware-targets-dvrs-forms-botnet/
+  - https://intezer.com/blog/research/kaiji-new-chinese-linux-malware-turning-to-golang/
+  - https://www.anomali.com/blog/illicit-cryptomining-threat-actor-rocke-changes-tactics-now-more-difficult-to-detect
+  - https://www.anomali.com/blog/pulling-linux-rabbit-rabbot-malware-out-of-a-hat
+  - https://blog.sucuri.net/2018/05/shell-logins-as-a-magento-reinfection-vector.html
+
+parameters:
+  - name: LinuxEnvGlobs
+    default: /home/*/.bashrc,/home/*/.bash_profile,/home/*/.bash_login,/home/*/.profile,/home/*/.zshrc,/etc/profile,/etc/environment,/home/*/.bash_logout
+    
+  - name: LoginScriptGlobs
+    default: /etc/profile.d/*.sh
+    
+  - name: LinuxEnvModifiers
+    default: ^(export|alias)
+    type: regex
+    
+  - name: LinuxEnvNetworkUtils
+    default: wget|curl|scp|ssh|nc\s|/usr/bin/nc\s|/bin/nc\s
+    type: regex
+    
+  - name: LinuxEnvScripting
+    default: python|perl|ruby|php|base64
+    type: regex
+    
+precondition: SELECT OS From info() where OS = 'linux'
+
+sources:
+    - name: ModifierDetection
+      query: |
+        LET EnvFiles = SELECT OSPath FROM glob(globs=split(string=LinuxEnvGlobs, sep=","))
+        SELECT * FROM foreach(row=EnvFiles,
+            query={
+                SELECT Line, OSPath FROM parse_lines(filename=OSPath)
+                WHERE
+                    Line =~ LinuxEnvModifiers
+            })
+
+    - name: NetworkUtilsDetection
+      query: |
+        LET EnvFiles = SELECT OSPath FROM glob(globs=split(string=LinuxEnvGlobs, sep=","))
+        SELECT * FROM foreach(row=EnvFiles,
+            query={
+                SELECT Line, OSPath FROM parse_lines(filename=OSPath)
+                WHERE
+                    Line =~ LinuxEnvNetworkUtils
+            })
+    
+    - name: ScriptingDetection
+      query: |
+        LET EnvFiles = SELECT OSPath FROM glob(globs=split(string=LinuxEnvGlobs, sep=","))
+        SELECT * FROM foreach(row=EnvFiles,
+            query={
+                SELECT Line, OSPath FROM parse_lines(filename=OSPath)
+                WHERE
+                    Line =~ LinuxEnvScripting
+            })
+            
+    - name: LoginScriptsDetection
+      query: |
+        LET ScriptFiles = SELECT OSPath FROM glob(globs=split(string=LoginScriptGlobs, sep=","))
+        SELECT * FROM foreach(row=ScriptFiles,
+            query={
+                SELECT * from read_file(filenames=OSPath)
```

Comment:
If you want to upload the file use the upload() function - do not read the file becuase that will put the data into the JSON